src/CreeDictionary/API/search/eip.py

```diff
@@ -0,0 +1,237 @@
+                new_tags=self.new_tags,
+            )
+        )
+
+    def inflect_search_results(self):
+        if not self.query_analyzed_ok:
+            return
+
+        inflected_results = self._generate_inflected_results()
+
+        # aggregating queries for performance
+        possible_wordforms = Wordform.objects.filter(
+            text__in=[r.inflected_text for r in inflected_results]
+        )
+        wordform_lookup = {}
+        for wf in possible_wordforms:
+            wordform_lookup[(wf.text, wf.lemma_id)] = wf
+
+        for result in inflected_results:
+            wordform = wordform_lookup.get(
+                (result.inflected_text, result.original_result.lemma_wordform.id)
+            )
+            if wordform is None:
+                # inflected form not found in DB, so create a synthetic one.
+                # This is unlikely to occur for Plains Cree, as it would require
+                # the EIP search to produce a valid analysis not covered by any
+                # paradigm file.
+                #
+                # Note: would not have auto-translations since those are
+                # currently only available for wordforms that were previously
+                # saved in the DB.
+                lemma = result.original_result.lemma_wordform
+
+                wordform = Wordform(
+                    text=result.inflected_text,
+                    lemma=lemma,
+                    inflectional_category=lemma.inflectional_category,
+                    pos=lemma.pos,
+                    stem=lemma.stem,
+                    analysis=result.analysis,
+                )
+
+            # if there are multiple inflections for the same original result, we
+            # may already have removed it
+            if self.search_run.has_result(result.original_result):
+                self.search_run.remove_result(result.original_result)
+
+            self.search_run.add_result(
+                result.original_result.create_related_result(
+                    wordform,
+                    is_eip_result=True,
+                )
+            )
+
+    def _generate_inflected_results(self) -> list[_EipResult]:
+        """
+        Of the results, sort out the verbs, then inflect them
+        using the new set of tags.
+        Return the inflected verbs.
+        """
+
+        words = []
+        for r in self.search_run.unsorted_results():
+            if "+V" in self.new_tags and r.is_lemma and r.lemma_wordform.pos == "V":
+                words.append(r)
+            if "+N" in self.new_tags and r.is_lemma and r.lemma_wordform.pos == "N":
+                words.append(r)
+
+        orig_tags_starting_with_plus: list[str] = []
+        tags_ending_with_plus: list[str] = []
+        for t in self.new_tags:
+            (
+                orig_tags_starting_with_plus
+                if t.startswith("+")
+                else tags_ending_with_plus
+            ).append(t)
```

Comment:
If this ternary seems unwieldy, it can be expanded to something like:

```suggestion
            if t.startswith("+"):
                tags = orig_tags_starting_with_plus
            else:
                tags = tags_ending_with_plus
            tags.append(t)
```

Reply:
This is an entirely fair suggestion. The ternary feels clever, but not readable.